packages/components/src/components/sidenavigation/sidenavigation.component.ts

```diff
@@ -0,0 +1,211 @@
+
+  /**
+   * Direction of the arrow icon, if applicable.
+   * - **positive**
+   * - **negative**
+   *
+   * Note: Positive and Negative directions are defined based on Cartesian plane.
+   * @default 'negative'
+   *
+   * @internal
+   */
+  @state()
+  private arrowDirection: Directions = this.flexibleExpanded ? DIRECTIONS.NEGATIVE : DIRECTIONS.POSITIVE;
+
+  protected override updated(changedProperties: Map<string, any>): void {
+    super.updated(changedProperties);
+
+    if (changedProperties.has('variant')) {
+      this.setVariant(this.variant);
+      this.updateExpansionState();
+    }
+
+    if (changedProperties.has('variant') || changedProperties.has('customerName')
+       || changedProperties.has('isExpanded')) {
+      this.updateContext();
+    }
+  }
+
+  /**
+   * Update all observing components of this
+   * provider to update the values
+   *
+   * Is called on every re-render, see Provider class
+   */
+  protected updateContext(): void {
+    if (this.context.value.variant !== this.variant
+        || this.context.value.customerName !== this.customerName
+        || this.context.value.isExpanded !== this.isExpanded
+    ) {
+      this.context.value.variant = this.variant;
+      this.context.value.customerName = this.customerName;
+      this.context.value.isExpanded = this.isExpanded;
+      this.context.updateObservers();
+    }
+  }
+
+  /**
+   * Syncs `isExpanded` and `aria-expanded` based on `variant` and `flexibleExpanded`.
+   *
+   * @internal
+   */
+  private updateExpansionState = (): void => {
+    if (this.variant === VARIANTS.FLEXIBLE) {
+      this.isExpanded = this.flexibleExpanded;
+      this.setAttribute('aria-expanded', String(this.flexibleExpanded));
+    } else {
+      this.isExpanded = this.variant === VARIANTS.FIXED_EXPANDED;
+      this.removeAttribute('aria-expanded');
+    }
+  };
+
+  /**
+   * Sets the variant attribute for the sideNavigation component.
+   * If the provided variant is not included in the variant,
+   * it defaults to the value specified in DEFAULTS.VARIANT.
+   *
+   * @param variant - The variant to set.
+   *
+   * @internal
+   */
+  private setVariant(variant: SideNavigationVariant) {
+    this.setAttribute('variant', Object.values(VARIANTS).includes(variant) ? variant : DEFAULTS.VARIANT);
+  }
+
+  /**
+   * Toggles navigation state for the flexible variant.
+   * Updates `flexibleExpanded`, `ariaExpanded`, and `arrowDirection`.
+   *
+   * @internal
+   */
+  private toggleSideNavigation(): void {
+    this.flexibleExpanded = !this.flexibleExpanded;
+    this.arrowDirection = this.arrowDirection === DIRECTIONS.NEGATIVE ? DIRECTIONS.POSITIVE : DIRECTIONS.NEGATIVE;
+    this.updateExpansionState();
+  }
+
+  public override render() {
+    return html`
+        <div part="side-navigation-container">
+          <div part="scrollable-section">
+            <slot name="scrollable-section"></slot>
+          </div>
```

Comment:
```suggestion
  public override render() {
     if (this.variant === VARIANTS.HIDDEN) {
      return html``;
    }
    return html`
        <div part="side-navigation-container">
          <div part="scrollable-section">
            <slot name="scrollable-section"></slot>
          </div>
```

Should we do this instead of setting 'display: none' on styling for 'hidden' variant?
This would be appropriate according to the docs mentioned (Removes the sidenavigation from the DOM.)

Reply:
Sure we can give it a try and verify that nothing breaks. It's important to ensure that the layout and internal state of child components are preserved without triggering re-creation.